rustls/tests/api.rs

```diff
@@ -5344,6 +5344,14 @@ fn connection_types_are_not_huge() {
     // Arbitrary sizes
     assert_lt(mem::size_of::<ServerConnection>(), 1600);
     assert_lt(mem::size_of::<ClientConnection>(), 1600);
+    assert_lt(
+        mem::size_of::<rustls::server::UnbufferedServerConnection>(),
+        1600,
+    );
+    assert_lt(
+        mem::size_of::<rustls::client::UnbufferedClientConnection>(),
+        1600,
+    );
```

Comment:
small nit: I think this change is unrelated to the stack exhaustion fix (though still a good idea!). Would you be willing to pull it out into a separate commit?